src/V8.rs

```diff
@@ -234,13 +234,16 @@ pub unsafe fn dispose() -> bool {
 }
 
 /// Clears all references to the v8::Platform. This should be invoked after
-/// V8 was disposed.
-pub fn shutdown_platform() {
+/// V8 was disposed. If it is called if V8 is not disposed, it will panic.
+pub fn dispose_platform() {
   let mut global_state_guard = GLOBAL_STATE.lock().unwrap();
-  // First shutdown platform, then drop platform
-  unsafe { v8__V8__ShutdownPlatform() };
+  // First check that the global state is disposed. If it is we dispose the
+  // platform. We then drop the platform.
   *global_state_guard = match *global_state_guard {
-    Disposed(_) => PlatformShutdown,
+    Disposed(_) => {
+      unsafe { v8__V8__DisposePlatform() };
+      PlatformShutdown
+    }
     _ => panic!("Invalid global state"),
   };
 }
```

Comment:
Please pay extra attention to this change. I updated the order of operations to be more correct. Previously we'd dispose the platform before checking if it is ready to be disposed. Now we check the global state before attempting disposal.

Reply:
Makes sense, though it appears that v8 already tracks a similar state-machine internally.

Do we wrap this only so we can throw rust-side panics rather than v8's `FATAL` ?

Reply:
I think so. @piscisaureus or @ry would have to confirm.